zqd/pcapstorage/pcapstorage.go

```diff
@@ -0,0 +1,210 @@
+package pcapstorage
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"sync"
+
+	"github.com/brimsec/zq/pcap"
+	"github.com/brimsec/zq/pcap/pcapio"
+	"github.com/brimsec/zq/pkg/iosrc"
+	"github.com/brimsec/zq/pkg/nano"
+	"github.com/brimsec/zq/zqd/api"
+	"github.com/brimsec/zq/zqe"
+)
+
+const (
+	MetaFile = "pcap.meta.json"
+)
+
+type Store struct {
+	meta meta
+	root iosrc.URI
+	mu   sync.Mutex
+}
+
+type meta struct {
+	PcapURI iosrc.URI
+	Span    nano.Span
+	Index   pcap.Index
+}
+
+type Info struct {
+	PcapURI  iosrc.URI
+	PcapSize int64
+	Span     nano.Span
+}
+
+func New(root iosrc.URI) *Store {
+	return &Store{root: root}
+}
+
+func Open(u iosrc.URI) (*Store, error) {
+	metauri := u.AppendPath(MetaFile)
+	b, err := iosrc.ReadFile(metauri)
+	if os.IsNotExist(err) {
+		return nil, zqe.E(zqe.NotFound, "%s: pcap store not found", metauri)
+	}
+	if err != nil {
+		return nil, err
+	}
+	var m meta
+	if err := json.Unmarshal(b, &m); err != nil {
+		return nil, err
+	}
+	return &Store{
+		root: u,
+		meta: m,
+	}, nil
+}
+
+func (s *Store) Rewrite(pcapuri iosrc.URI) error {
```

Comment:
Consider 'Update' ; unlike the file store, we're not overwriting the bulk data (the pcap itself).

```suggestion
func (s *Store) Update(pcapuri iosrc.URI) error {
```

Reply:
Ok.